src/libmongoc/doc/mongoc_client_get_handshake_description.rst

```diff
@@ -0,0 +1,50 @@
+:man_page: mongoc_client_get_handshake_description
+
+mongoc_client_get_handshake_description()
+=========================================
+
+Synopsis
+--------
+
+.. code-block:: c
+
+   mongoc_server_description_t *
+   mongoc_client_get_handshake_description (mongoc_client_t *client,
+                                            uint32_t server_id,
+                                            bson_t *opts,
+                                            bson_error_t *error);
+
+Returns a description constructed from the initial handshake response to a server.
+
+Description
+-----------
+
+:symbol:`mongoc_client_get_handshake_description` is distinct from :symbol:`mongoc_client_get_server_description`. :symbol:`mongoc_client_get_server_description` returns a server description constructed from monitoring, which may differ from the server description constructed from the connection handshake.
+
+:symbol:`mongoc_client_get_handshake_description` does not attempt to establish a connection to the server if a connection was not already established. It will complete authentication on a single-threaded client if the connection has not yet been used by an operation.
+
+To establish a connection on a single-threaded client, ensure the server is selectable (the monitoring connection is the only connection made to the server). This can be done with a function like :symbol:`mongoc_client_select_server`, or any function which performs server selection.
```

Comment:
> If this is going to be difficult for wrapping drivers to utilize. It may be simpler for this to always attempt to establish a connection if one does not exist. That way, callers do not need to do any special checking to ensure a connection is established.

I do think having `mongoc_client_get_handshake_description` establish a connection would be desirable. I can't think of a reason why we'd actually want an empty server description returned (unless the connection could not be established in which case it'd likely be an error).

Reply:
That sounds good to me. I will proceed with that change. My original thinking was to have this function do no more than necessary. But ensuring a connection is established seems more awkward.